woocommerce/woocommerce
# Add cherry pick tool

### All Submissions:






### Changes proposed in this Pull Request:



This PR adds a utility tool to automate cherry picking steps that release leads often do after a release branch has already been cut. Note that currently it doesn't handle errors you might get from running git commands such as conflicts, server errors..etc. This is very much just a MVP.

`Usage: pnpm cherry-pick <release_branch> <pull_request_number>. Separate pull request numbers with a comma (no space) if more than one.`

### How to test the changes in this Pull Request:

1. Create a personal access token with first checkbox checked (Repo) and assign it to environment variable `GITHUB_CHERRY_PICK_TOKEN`.
2. Depending if you're using SSH or HTTPS for GitHub, set the environment variable `GITHUB_REMOTE_URL` to either `https` or `ssh`.
3. Checkout this PR and run `pnpm install`.
4. For the test you can run `pnpm cherry-pick release/6.7 33676,33716`.
5. Check that all the processing steps in the terminal has a green checkmark and nothing failed with a red checkmark.
6. Ensure two PRs are created. One that targets the release branch that has the cherry picks and another that targets trunk which has deleted changelog files.
7. Review the PRs to ensure the commits look right.
8. Remember to closed the two test PRs and delete the associated branches that it created.

### Other information:




### FOR PR REVIEWER ONLY:

-
